qemu/tests/uefi_check_debugcon.py

```diff
@@ -0,0 +1,118 @@
+import logging
+
+from virttest import utils_test
+from virttest import utils_misc
+from virttest import env_process
+from virttest import error_context
+from avocado.utils import process
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Verify the "-debugcon" parameter under the UEFI environment:
+    1) Boot up a guest.
+       If params["ovmf_log"] is not None,
+       append debugcon parameter to qemu command lines.
+    2) Remove the existing isa-log device.
+    3) Destroy the guest.
+    4) Start the trace command on host.
+    5) Re-create the guest and verify it is alive.
+    6) Destroy the guest.
+    7) Check pio_read counts and pio_write counts.
+    7.1) If disable debugcon:
+            pio_read_counts > 0
+            pio_write_counts = 0
+    7.2) If enable debugcon:
+            pio_read_counts > 0
+            pio_write_counts > 0
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def check_trace_process():
+        """
+        check whether trace process is existing
+        """
+        if process.system(
+                params["grep_trace_cmd"], ignore_status=True, shell=True):
+            return False
+        else:
+            return True
+
+    def remove_isa_debugcon(vm):
+        """
+        remove the existing isa-log device
+        """
+        for device in vm.devices:
+            if device.type == "isa-log":
+                vm.devices.remove(device)
+                break
+        env.register_vm(vm.name, vm)
+
+    def trace_kvm_pio():
+        """
+        trace event kvm_pio
+        """
+        process.system(trace_record_cmd)
+
+    error_context.context("Check whether trace-cmd has been "
+                          "installed on host.", logging.info)
+    if process.system("which trace-cmd", ignore_status=True):
```

Comment:
cmds_installed_host += 'trace-cmd', so the check here can be removed, just fyi.

Reply:
I think all work well. thank you.